File: woodroid/src/main/java/me/gilo/woodroid/data/api/TaxRateAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.TaxRate;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public interface TaxRateAPI {

    @Headers("Content-Type: application/json")
    @POST("taxes")
    Call<TaxRate> create(@Body TaxRate body);

    @GET("taxes/{id}")
    Call<TaxRate> view(@Path("id") int id);

    @GET("taxes")
    Call<List<TaxRate>> list();

    @Headers("Content-Type: application/json")
    @PUT("taxes/{id}")
    Call<TaxRate> update(@Path("id") int id, @Body TaxRate body);

    @DELETE("taxes/{id}")
    Call<TaxRate> delete(@Path("id") int id);

    @DELETE("taxes/{id}")
    Call<TaxRate> delete(@Path("id") int id, @Query("force") boolean force);

    @POST("taxes/batch")
    Call<String> batch(@Body TaxRate body);

    @GET("coupons")
    Call<ArrayList<TaxRate>> filter(@QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/TaxRateAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.TaxRate
import retrofit2.Call
import retrofit2.http.*

import java.util.ArrayList

interface TaxRateAPI {

    @Headers("Content-Type: application/json")
    @POST("taxes")
    fun create(@Body body: TaxRate): Call<TaxRate>

    @GET("taxes/{id}")
    fun view(@Path("id") id: Int): Call<TaxRate>

    @GET("taxes")
    fun list(): Call<List<TaxRate>>

    @Headers("Content-Type: application/json")
    @PUT("taxes/{id}")
    fun update(@Path("id") id: Int, @Body body: TaxRate): Call<TaxRate>

    @DELETE("taxes/{id}")
    fun delete(@Path("id") id: Int): Call<TaxRate>

    @DELETE("taxes/{id}")
    fun delete(@Path("id") id: Int, @Query("force") force: Boolean): Call<TaxRate>

    @POST("taxes/batch")
    fun batch(@Body body: TaxRate): Call<String>

    @GET("coupons")
    fun filter(@QueryMap filter: Map<String, String>): Call<ArrayList<TaxRate>>

}